flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/WatermarkSpec.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api;
+
+import org.apache.flink.table.types.DataType;
+
+import java.util.Objects;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Watermark metadata defined in {@link TableSchema}. It mainly contains two parts,
+ * 1) the rowtime attribute.
+ * 2) the watermark generation strategy.
+ */
+public class WatermarkSpec {
+
+	private final String rowtimeAttribute;
+
+	private final String watermarkExpressionString;
+
```

Comment:
Can we just comment all these members ?

Reply:
I can copy the javadoc to the members here, but I think it is not necessary. Because the javadoc of the getter methods explain them in detail. These private members are invisible to users.